pkg/e2e/operators/cloudingress/deletetapps2.go

```diff
@@ -0,0 +1,123 @@
+package cloudingress
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"time"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	cloudingressv1alpha1 "github.com/openshift/cloud-ingress-operator/pkg/apis/cloudingress/v1alpha1"
+	"github.com/openshift/osde2e/pkg/common/constants"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/e2e/operators"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ = ginkgo.Describe(constants.SuiteInforming+TestPrefix, func() {
+	h := helper.New()
+	ginkgo.Context("Delete apps2 ingresscontroller", func() {
```

Comment:
Thanks for making all those changes @lnguyen1401 , appreciated! They all look good to me.

I just want to revisit one note from my earlier review around the number of `time.Sleep`s occuring. This test in particular is pretty heavy on them, there's a 30s sleep, a 60s sleep, a 2 minute sleep to delete the ingress controller, a ten minute sleep after that function finishes, then an optional 2 minute sleep again after that. Lots of sleepyness, especially in the case of upgrade jobs, where this test would run twice. :) 

The main issue with this is not just the test doing a lot of nothing for a lof of the time, but more that it eats into the overall job's duration on Prow (which is limited), so wherever we can it's good to get this tighter. 

Is it possible to detect on-cluster the sorts of conditions you're expecting to have occurred after the sleeps and poll for that condition periodically?

ie. Here is how I would consider refactoring the `removeIngressController` to not have a sleep (disclaimer: not tested!):

```
func removeIngressController(h *helper.H, name string) error {
	_, exists, index := appIngressExits(h, false, name)
	// only remove the secondary ingress if it already exist in the publishing strategy
	if exists {
		removeAppIngress(h, index)
	}
	err = wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
		_, err := h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Get(context.TODO(), ingressControllerName, metav1.GetOptions{})
		if k8serrors.IsNotFound(err) {
			return true, nil
		}
		return false, err
	})
	return err
}
```

and then back in the calling code:

```
			err = removeIngressController(h, ingressControllerName)
			Expect(err).NotTo(HaveOccurred()
```

Totally up to you if you want to explore this (I'm happy to help more if you're interested). Otherwise I'm okay to lgtm it personally.

Reply:
Yeah I think this would be a good thing to do